news/7555.removal

```diff
@@ -0,0 +1,10 @@
+Building of local directories is now done in place. Previously
+pip did copy the local directory tree to a temporary location before
+building. That approach had a number of drawbacks, among which performance
+issues, as well as various issues arising when the python project directory
+depends on it's parent directory (such as the presence of a VCS directory).
+The user visible effect of this change is that secondary build artifacts,
+if any, may therefore be created in the local directory, whereas before they
+were created in a temporary copy of the directory and then deleted.
+This notably includes the ``.egg-info`` directory in the case of the
```

Comment:
With #7731 the `.egg-info` note isn't relevant anymore ?

Reply:
@xavfernandez .egg-info will should disturb pip anymore, but it is still generated by setup.py builds so I think it is still worth mentioning. I added a mention of directory `build` that is also generated by setup.py.